robosuite/environments/manipulation/nut_assembly.py

```diff
@@ -468,83 +576,23 @@ def _reset_internal(self):
-
-            `'robot-state'`: contains robot-centric information.
-
-            `'object-state'`: requires @self.use_object_obs to be True. Contains object-centric information.
-
-            `'image'`: requires @self.use_camera_obs to be True. Contains a rendered frame from the simulation.
-
-            `'depth'`: requires @self.use_camera_obs and @self.camera_depth to be True.
-            Contains a rendered depth map from the simulation
-
-        Returns:
-            OrderedDict: Observations from the environment
-        """
-        di = super()._get_observation()
-
-        # low-level object information
-        if self.use_object_obs:
-            # Get robot prefix
-            pr = self.robots[0].robot_model.naming_prefix
-
-            # remember the keys to collect into object info
-            object_state_keys = []
-
-            # for conversion to relative gripper frame
-            gripper_pose = T.pose2mat((di[pr + "eef_pos"], di[pr + "eef_quat"]))
-            world_pose_in_gripper = T.pose_inv(gripper_pose)
-
-            for i, nut in enumerate(self.nuts):
-
-                if self.single_object_mode == 2 and self.nut_id != i:
-                    # skip observations
-                    continue
-
-                obj_str = nut.name
-                obj_pos = np.array(self.sim.data.body_xpos[self.obj_body_id[obj_str]])
-                obj_quat = T.convert_quat(
-                    self.sim.data.body_xquat[self.obj_body_id[obj_str]], to="xyzw"
-                )
-                di["{}_pos".format(obj_str)] = obj_pos
-                di["{}_quat".format(obj_str)] = obj_quat
-
-                object_pose = T.pose2mat((obj_pos, obj_quat))
-                rel_pose = T.pose_in_A_to_pose_in_B(object_pose, world_pose_in_gripper)
-                rel_pos, rel_quat = T.mat2pose(rel_pose)
-                di["{}_to_{}eef_pos".format(obj_str, pr)] = rel_pos
-                di["{}_to_{}eef_quat".format(obj_str, pr)] = rel_quat
-
-                object_state_keys.append("{}_pos".format(obj_str))
-                object_state_keys.append("{}_quat".format(obj_str))
-                object_state_keys.append("{}_to_{}eef_pos".format(obj_str, pr))
-                object_state_keys.append("{}_to_{}eef_quat".format(obj_str, pr))
-
-            if self.single_object_mode == 1:
-                # zero out other objs
-                for obj in self.model.mujoco_objects:
-                    if obj.name == self.obj_to_use:
-                        continue
-                    else:
-                        di["{}_pos".format(obj.name)] *= 0.0
-                        di["{}_quat".format(obj.name)] *= 0.0
-                        di["{}_to_{}eef_pos".format(obj.name, pr)] *= 0.0
-                        di["{}_to_{}eef_quat".format(obj.name, pr)] *= 0.0
-
-            di["object-state"] = np.concatenate([di[k] for k in object_state_keys])
-
-        return di
+        # Make sure to update sensors' active and enabled states
```

Comment:
Enabled = The observable will internally actually update its value and add its result to the `obs_cache`.

Active = The observable output will actually be returned in the obs_dict from `_get_observations()`.

The main distinction is that some observables should be internal but not necessarily required / desired for the actual outputted observation dictionary. A concrete use case is the `joint_pos` observable vs. the `joint_pos_cos`, `joint_pos_sin` observables. The former is enabled but NOT active since its outputted value is required to calculate the corresponding cos / sin values, but not desired for the actual outputted `obs_dict` since it contains redundant information relative to the cos / sin values.

Reply:
see https://github.com/ARISE-Initiative/robosuite/pull/146/files#diff-1e7090b536b9ec1f865375071b2667c19aad82b6250a147d25d21d8930df92c1R238 and https://github.com/ARISE-Initiative/robosuite/pull/146/files#diff-1e7090b536b9ec1f865375071b2667c19aad82b6250a147d25d21d8930df92c1R250